File: LeetCode/Arrays/217 Contains Duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {

    // Sorting the array first using quicksort
    nums = nums.sort()

    for (let i = 0; i < nums.length; i++) {
        if (nums[i] == nums[i + 1]) {
            return true;
        }
    }
    return false;

    // Time Complexity: O(nlogn) 
    // Space Complexity: O(1)
    
};

File: 2026/Blind 75 [Python]/Arrays & Hashing/217.Contains Duplicate.py
from typing import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        numSet = set(nums)

        if len(numSet) != len(nums):
            return True
        else:
            return False


# Test cases
solution = Solution()

print(solution.containsDuplicate([1, 2, 3, 1]))  # True
print(solution.containsDuplicate([1, 2, 3, 4]))  # False
print(solution.containsDuplicate([1, 1]))        # True
print(solution.containsDuplicate([]))            # False


# Time Complexity: O(N)
# Space Complexity: O(N)